src/contract/assignments.rs

```diff
@@ -778,13 +842,37 @@ where
     }
 }
 
-impl<STATE> CommitEncodeWithStrategy for OwnedState<STATE>
+impl<STATE> CommitEncode for OwnedState<STATE>
 where
+    Self: Clone,
     STATE: StateTypes,
     STATE::Confidential: PartialEq + Eq,
     STATE::Confidential: From<<STATE::Revealed as Conceal>::Confidential>,
+    STATE::Revealed: CommitEncode,
+    STATE::Confidential: CommitEncode,
 {
-    type Strategy = commit_strategy::UsingConceal;
+    fn commit_encode<E: io::Write>(mut self, mut e: E) -> usize {
+        self = self.conceal_seal().conceal();
```

Comment:
I feel like this extra step of `conceal_seal()` is redundant. The output of `conceal_seal()` will always be either `ConfidentialSeal` or `Confidential`, so the below `Reveled` and `ConfidentailAmount` code branch will never hit. 

Instead as per the discussion in the last call, we should modify the `conceal()` definition of `OwnedState` to conceal the correct parts. Which is not being done currently. 

Then the `commit_encode()` definition can simply use the `UsingConceal` strategy, which should then give the correct results.

cc: @dr-orlovsky for the modification of the `conceal()` operation. My guess is it should always return `Confidential. This is explicitly being done now in this PR with the `commit_encode()` and `conceal_seal()` method, but it can be achieved implicitly with modification of `conceal()` method.

Reply:
> I feel like this extra step of `conceal_seal()` is redundant. The output of `conceal_seal()` will be either `ConfidentialSeal` or `Confidential` and applying another `conceal()` will always result in a `Confidential` state, so the below `Reveled`, `ConfidentialSeal` and `ConfidentailAmount` code branch will never hit.
> 
> Instead as per the discussion in the last call, we should modify the `conceal()` definition of `OwnedState` to conceal the correct parts. Which is not being done currently.

Your analysis (as per the paragraphs above) is correct. The `match` statement should indeed have only one active branch for the `OwnedState::Confidential` case.

As for the use of a separate `conceal_seal()` method, I did it that way because I was not sure if the `conceal()` method could be changed (to conceal not only the value, as it does today, but also the seal) without breaking any other part of the code.

So, if you guys tell me that I could simply change the `conceal()` method instead, I will make the change.

Reply:
> Then the `commit_encode()` definition can simply use the `UsingConceal` strategy, which should then give the correct results.

Unfortunately, the above statement is not correct. We need to explicitly implement the `CommitEncode` trait for the `OwnedState<STATE>` type for the same reason why we needed to do it for the `Void` type. Please see my comments [above](https://github.com/rgb-org/rgb-core/pull/6#discussion_r577886073).